sage_modabvar/abvar.py

```diff
@@ -583,6 +585,58 @@ def newform_label(self):
             group = 'GH%s'%(str(G._generators_for_H()).replace(' ',''))
         return '%s%s%s'%(N, cremona_letter_code(self.isogeny_number()), group)
 
+    def elliptic_curve(self):
+        """
+        Return an elliptic curve isogenous to self. If self is not dimension 1
+        with rational base ring, raise a ValueError.
+
+        The elliptic curve is found by looking it up in Cremona's tables.
+
+        OUTPUT: an elliptic curve isogenous to self.
+
+        EXAMPLES::
+
+            sage: from sage_modabvar import J0
+            sage: J = J0(11)
+            sage: J.elliptic_curve()
+            Elliptic Curve defined by y^2 + y = x^3 - x^2 - 10*x - 20 over Rational Field
+
+            sage: J = J0(49)
+            sage: J.elliptic_curve()
+            Elliptic Curve defined by y^2 + x*y = x^3 - x^2 - 2*x - 1 over Rational Field
+
+            sage: A = J0(37)[1]
+            sage: E = A.elliptic_curve()
+            sage: A.lseries()(1)
+            0.725681061936153
+            sage: E.lseries()(1)
+            0.725681061936153
+
+        """
+
+        if self.dimension() > 1:
+            raise ValueError("self must be of dimension 1")
+        if self.base_ring() != QQ:
+            raise ValueError("base ring must be QQ")
+
+        f = self.newform('a')
+        N = f.level()
+
+        c = CremonaDatabase()
+        if N > c.largest_conductor():
+            raise RuntimeError("Elliptic curve not found" +
```

Comment:
Oh good point.  Mention he behavior in the doc string and point out that in
practice user will never see it due to performance.

On Tuesday, July 19, 2016, Kevin Lui <notifications@github.com> wrote:

> In sage_modabvar/abvar.py
> <https://github.com/williamstein/sage_modabvar/pull/28#discussion_r71417328>
> :
>
> > +            sage: E.lseries()(1)
> > +            0.725681061936153
> > +
> > +        """
> > +
> > +        if self.dimension() > 1:
> > +            raise ValueError("self must be of dimension 1")
> > +        if self.base_ring() != QQ:
> > +            raise ValueError("base ring must be QQ")
> > +
> > +        f = self.newform('a')
> > +        N = f.level()
> > +
> > +        c = CremonaDatabase()
> > +        if N > c.largest_conductor():
> > +            raise RuntimeError("Elliptic curve not found" +
>
> How do I do this?
>
> I actually haven't test this code branch. It takes Sage a really long time
> to create an abvar of such large conductor.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/williamstein/sage_modabvar/pull/28/files/cecf1fee128218e7aae3976c3f1fb7bd044e5f2b#r71417328>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABN5dpOPMf82RPDNg6FVfKEoK5QYZ6Vzks5qXTQlgaJpZM4JQGcJ>
> .
>


-- 
Sent from my massive iPhone 6 plus.